tempodb/encoding/vparquet/create.go

```diff
@@ -38,7 +38,7 @@ func CreateBlock(ctx context.Context, cfg *common.BlockConfig, meta *backend.Blo
 	s := newStreamingBlock(ctx, cfg, meta, r, to, tempo_io.NewBufferedWriter)
 
 	for {
-		_, obj, err := i.Next(ctx)
+		id, obj, err := i.Next(ctx)
```

Comment:
this iterator is just exposing slices over an internal buffer. if id is held onto longer than this loop we should make a copy.

Reply:
Good catch. It looks like it's actually ok since inside the iterator is using fresh byte slices in [UnmarshalObjectFromReader](https://github.com/grafana/tempo/blob/main/tempodb/encoding/v2/object.go#L64), but this is also unnecessarily hanging onto the object body.  Will fix.